block/block_cache.go

```diff
@@ -0,0 +1,45 @@
+package block
+
+import (
+	"sync"
+
+	"github.com/dymensionxyz/dymint/types"
+)
+
+type Cache struct {
+	cache map[uint64]CachedBlock
+	sync.Mutex
+}
+
+func (m *Cache) AddBlockToCache(h uint64, b *types.Block, c *types.Commit) {
+	m.Lock()
+	defer m.Unlock()
+	m.cache[h] = CachedBlock{Block: b, Commit: c}
+	types.BlockCacheSizeGauge.Set(float64(len(m.cache)))
+}
```

Comment:
func (m *Cache) AddBlockToCache(h uint64, b *types.Block, c *types.Commit) {
	m.Lock()
	m.cache[h] = CachedBlock{Block: b, Commit: c}
	m.Unlock()
	m.SetSizeMetric()
}

func (m *Cache) SetSizeMetric() {
	types.BlockCacheSizeGauge.Set(float64(m.Size()))
}

is preferable I think, you should reuse Size() if you implemented it